### ISSUE TITLE
DeepChem 2.6.0 Release Blocker Checklist

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#2512]
We're getting close to DeepChem 2.6.0 (code freeze for PRs is this coming Friday the 14th) so I'm starting up this list to track some release blocking bugs we will need to fix before 2.6.0:

- [ ] https://github.com/deepchem/deepchem/issues/2490: Mysterious failures on book code, chapter 11 with DeepChem head
- [ ] https://github.com/deepchem/deepchem/issues/2477: Potentially serious weave performance drops. To test, we'd need to train a decent sized weave model on a molnet benchmark to see if performance has dropped considerable.
- [x] https://github.com/deepchem/deepchem/issues/2511: dgllife installation errors
- [ ] Running the full test suite: (Ideally, we should have a way to kick off a full test suite run automatically but for now this has to be done manually)

**Comment by rbharath** [comment]
I've been out for the last few days recovering from my second vaccine dose but now starting to come back online. We're now in code freeze till the release. I'll be working to get the release out the door over the next few days (starting tomorrow)

**Comment by rbharath** [comment]
Running the full slow test suite now. Will log issues I'm seeing here:

- Should we add python 3.8 support in this release? Feels like a lot of people are on 3.8 or even 3.9 by default now
- pdbfixer doesn't support python 3.8 which is one issue though (CC @peastman)

**Comment by peastman** [comment]
PDBFixer is a noarch package.  It can work on any Python >= 3.6.  The only special dependency it has is OpenMM, and that has packages all the way up to 3.9.

Yes, we should definitely support 3.8.  It was released all the way back in 2019!  And 3.9 too if we can.  It was released over six months ago.

**Comment by rbharath** [comment]
Sounds good on pdbfixer! I'll make a PR later doing a bump to 3.9/3.9 after I finish a local runthrough of the test suite on my system

Another issue:
- mdtraj only supports up to python 3.6 looks like? Conda refused to install for me and when I did the pip install I got a scary error about C incompatibilities

**Comment by rbharath** [comment]
Running through the slow tests, here's one failing test:

```
_______________________________________ TestMol2VecFingerprint.test_mol2vec_fingerprint ________________________________________

self = <deepchem.feat.tests.test_mol2vec_fingerprint.TestMol2VecFingerprint testMethod=test_mol2vec_fingerprint>

    def test_mol2vec_fingerprint(self):
      """
      Test simple fingerprint.
      """
      featurizer = Mol2VecFingerprint()
      feature = featurizer([self.mol])
>     assert feature.shape == (1, 300)
E     AssertionError: assert (1, 0) == (1, 300)
E       At index 1 diff: 0 != 300
E       Use -v to get the full diff

deepchem/feat/tests/test_mol2vec_fingerprint.py:25: AssertionError
```
I'm not sure what's going on here. Will try to capture all failing tests then fix

**Comment by rbharath** [comment]
Another repeatedly failing test:
```
======================================================================
FAIL: test_hindsight (deepchem.rl.tests.test_a2c.TestA2C)
Test Hindsight Experience Replay.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/bharath/Code/deepchem/deepchem/rl/tests/test_a2c.py", line 251, in test_hindsight
    assert pass_count >= 3
AssertionError
```
I'm not sure if this one is flaky. I tried running it a few times and got the error over and over though.

**Comment by rbharath** [comment]
As a quick update on this thread I'm finding it a little hard to do much work this week (getting married at the end of the week), so while I've been making progress on fixing tests I don't think I'll be able to get this out this week. I'm going to be offline until mid next week for the wedding but will pick up release work as soon as I'm back online :)

**Comment by rbharath** [comment]
Fixed a couple of tests in #2531 but I think there are a few more still failing. Will continue doing my run through

**Comment by rbharath** [comment]
I'm seeing some failures in #2531 that I think are caused by the fact that the latest versions of pytorch require numpy >= 1.20.1 which isn't available in python 3.6.

Would folks be OK with us dropping python 3.6 support?

**Comment by peastman** [comment]
I'd be ok with dropping Python 3.6, but upgrading numpy will introduce a lot of mypy failures that we then need to fix.  See #2382.

**Comment by rbharath** [comment]
One additional bit of news: TF 2.5.0 is now out https://github.com/tensorflow/tensorflow/releases/tag/v2.5.0 with python 3.9 support

**Comment by rbharath** [comment]
Updating this issue to reflect the discussion in https://github.com/deepchem/deepchem/pull/2531. As a summary, since TF doesn't yet support numpy 1.20, we are blocked on releasing DeepChem 2.6.0 since we have many TF models which break on numpy 1.20 and pytorch/dgl models which break on numpy 1.19. Given this state of affairs, we can plan to fix everything except the known TF breakages and then make the DeepChem 2.6.0 release once TF has upgraded to numpy 1.20.

As a general comment, I'll say TensorFlow seems to be our biggest cause of breakages. We had major issues with TF 2, issues with TF 2.4.0, and now issues because of TF/numpy-1.20. Perhaps we should consider shifting more of our models to PyTorch and reducing our TF footprint over time.
